custom_components/switchbot-curtain/config_flow.py

```diff
@@ -0,0 +1,93 @@
+"""Config flow for Switchbot."""
+import logging
+
+import pygatt
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_MAC, CONF_NAME, CONF_PASSWORD, CONF_SENSOR_TYPE
+
+from .const import ATTR_BOT, ATTR_CURTAIN, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _btle_connect(mac):
+    """Try to connect to switchbot device."""
+    adapter = pygatt.GATTToolBackend()
+
+    try:
+        adapter.start()
+        device = adapter.connect(mac, address_type=pygatt.BLEAddressType.random)
+    finally:
+        adapter.stop()
+
+    return device
+
+
```

Comment:
Found a potential problem here. Please wait while I fix the device connection test.

Reply:
-Fixed invalid command error with adapter.start(reset_on_start=False)
-Added thread lock as multiple connections could cause issues with pygatt.